src/app/dataset-view/additional-components/overview-component/components/readme-section/readme-section.component.ts

```diff
@@ -124,7 +124,7 @@ export class ReadmeSectionComponent extends BaseComponent implements OnChanges,
                     linkElement.setAttribute("target", "_blank");
                     linkElement.setAttribute(
                         "href",
-                        `/${this.datasetBasics.owner.accountName}/${this.datasetBasics.name}?tab=${DatasetViewTypeEnum.Data}&sqlQuery=${sqlQueries[index]}`,
+                        `/${this.datasetBasics.owner.accountName}/${this.datasetBasics.name}?tab=${DatasetViewTypeEnum.Data}&sqlQuery=${encodeURI(sqlQueries[index])}`,
```

Comment:
Let's have some test, since we had a bug!

Reply:
Added unit test, done